channeldb/addr_source.go

```diff
@@ -10,8 +11,9 @@ import (
 // node. It may combine the results of multiple address sources.
 type AddrSource interface {
 	// AddrsForNode returns all known addresses for the target node public
-	// key.
-	AddrsForNode(nodePub *btcec.PublicKey) ([]net.Addr, error)
+	// key. The returned boolean must indicate if the given node is unknown
+	// to the backing source.
+	AddrsForNode(nodePub *btcec.PublicKey) (bool, []net.Addr, error)
```

Comment:
> since the caller can know if this is known or not by checking the error

Not necessarily since there are possibly multiple backends. The caller should not need to know to check both `graphdb.NotFound` and `channeldb.NotFound`. 

> ut I guess it's nice to have this abstraction so the caller doesn't need to care about the specific errors.

yeah exactly

> if we decide to use the intersection instead of the union of the sets, we can then remove this bool.

I think we want the union though for maximum address coverage. The reason we check multiple sources is so that we have a higher probability of using a more up to date address